File: problems/230. Kth Smallest Element in a BST/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        inOrder = []
        self.getElements(root, inOrder)
        return inOrder[k-1]

    def getElements(self, node, arr):
        if not node:
            return
        self.getElements(node.left, arr)
        arr.append(node.val)
        self.getElements(node.right, arr)

File: problems/230. Kth Smallest Element in a BST/solution.cpp
class Solution
{
public:
    int kthSmallest(TreeNode *root, int k)
    {
        if (root == NULL)
            return -1;
        std::vector<int> inOrder;
        getElements(root, inOrder);
        return inOrder[k - 1];
    }

    void getElements(TreeNode *node, std::vector<int> &arr)
    {
        if (node == NULL)
            return;
        getElements(node->left, arr);
        arr.push_back(node->val);
        getElements(node->right, arr);
    }
};